apps/meteor/server/services/video-conference/service.ts

```diff
@@ -1034,4 +1070,124 @@ export class VideoConfService extends ServiceClassInternal implements IVideoConf
+		if (!(await this.isPersistentChatEnabled())) {
+			return;
+		}
+
+		const call = await VideoConferenceModel.findOneById(callId, {
+			projection: { rid: 1, createdBy: 1, discussionRid: 1, providerName: 1 },
+		});
+		if (!call) {
+			throw new Error('invalid-video-conference');
+		}
+
+		// If there's already a discussion assigned to it, do not create a new one
+		if (call.discussionRid) {
+			return;
+		}
+
+		// If the call provider does not explicitly support persistent chat, do not create discussions
+		if (!videoConfProviders.getProviderCapabilities(call.providerName)?.persistentChat) {
+			return;
+		}
+
+		const name = settings.get<string>('VideoConf_Persistent_Chat_Discussion_Name') || i18n.t('Conference Call Chat History');
+		const displayName = `${name} - ${new Date().toISOString().substring(0, 10)}`;
+
+		await this.createDiscussionForConference(displayName, call, createdBy);
+	}
+
+	private async getRoomForDiscussion(
+		baseRoom: IRoom['_id'],
+		childRoomIds: IRoom['_id'][] = [],
+	): Promise<Pick<IRoom, '_id' | 't' | 'teamId' | 'prid'>> {
+		const room = await Rooms.findOneById<Pick<IRoom, '_id' | 't' | 'teamId' | 'prid'>>(baseRoom, {
+			projection: { t: 1, teamId: 1, prid: 1 },
+		});
+		if (!room) {
+			throw new Error('invalid-room');
+		}
+
+		if (room.prid) {
+			if (childRoomIds.includes(room.prid)) {
+				throw new Error('Room has circular reference.');
+			}
+
+			return this.getRoomForDiscussion(room.prid, [...childRoomIds, room._id]);
+		}
+
+		return room;
+	}
+
+	private async createDiscussionForConference(
+		name: string,
+		call: AtLeast<VideoConference, '_id' | 'rid' | 'createdBy'>,
+		createdBy?: IUser,
+	): Promise<void> {
+		const room = await this.getRoomForDiscussion(call.rid);
+
+		const type = await roomCoordinator.getRoomDirectives(room.t).getDiscussionType(room);
+		const user = call.createdBy._id === createdBy?._id ? createdBy : await Users.findOneById(call.createdBy._id);
+		if (!user) {
+			throw new Error('invalid-user');
+		}
+
+		const discussion = await createRoom(
+			type,
+			Random.id(),
+			user,
+			[],
+			false,
+			false,
+			{
+				fname: name,
+				prid: room._id,
+				encrypted: false,
+			},
+			{
+				creator: user._id,
+			},
+		);
+
+		return this.assignDiscussionToConference(call._id, discussion._id);
+	}
+
+	public async assignDiscussionToConference(callId: VideoConference['_id'], rid: IRoom['_id'] | undefined): Promise<void> {
+		// Ensures the specified rid is a valid room
+		const room = rid ? await Rooms.findOneById<Pick<IRoom, '_id' | 'prid'>>(rid, { projection: { prid: 1 } }) : null;
+		if (rid && !room) {
+			throw new Error('invalid-room-id');
+		}
+
+		const call = await VideoConferenceModel.findOneById(callId, { projection: { users: 1, messages: 1 } });
+		if (!call) {
+			return;
+		}
+
+		await VideoConferenceModel.setDiscussionRidById(callId, rid);
+
+		if (room) {
+			await Promise.all(call.users.map(({ _id }) => this.addUserToDiscussion(room._id, _id)));
```

Comment:
wonder if this will present any problem on big calls (with a ton of participants). Would it be a premature opt to batch these in groups instead of sending all them at once?

Reply:
I wondered about this as well but with the current use cases this function will always be called with an empty list of users and I was just adding this here for compatibility in case an app is changed down the line. I didn't want to over engineer it for something that may never really be needed.

Reply:
interesting, i thought this was kinda an "add all users from call to discussion" thing, if it's not something that will be used that often or with many users, then should be fine